src/network/requestQueue/socketRequest.js

```diff
@@ -0,0 +1,116 @@
+const { KafkaJSRequestTimeoutError } = require('../../errors')
+
+const PRIVATE = {
+  STATE: Symbol('private:SocketRequest:state'),
+}
+
+const REQUEST_STATE = {
+  PENDING: Symbol('PENDING'),
+  SENT: Symbol('SENT'),
+  COMPLETED: Symbol('COMPLETED'),
+  REJECTED: Symbol('REJECTED'),
+}
+
+/**
+ * @typedef {Object} SocketRequest
```

Comment:
Might be nice to add a line explaining the purpose of this class and why it's needed

Reply:
8b1173d